cipher_suite.go

```diff
@@ -189,18 +193,26 @@ func parseCipherSuites(userSelectedSuites []CipherSuiteID, excludePSK, excludeNo
 		cipherSuites = defaultCipherSuites()
 	}
 
+	var foundPSK, foundNonPSK bool
```

Comment:
For symmetry I'd prefer `foundPKI` instead of `foundNonPSK`

Reply:
Do you think it'd be best to universally switch the "nonPSK" nomenclature to `PKI` (e.g. `includeNonPSK`, `errNoAvailableNonPSKCipherSuite`)? I just subconsciously used the original wording from the `excludeNonPSK` parameter name, but I think that changing them might be reasonable too; my only concern would be that the RFCs tend to refer to "PSK cipher suites" but don't qualify other cipher suites as "PKI cipher suites"; I'm a fairly new reader of them and could just be overthinking it, though. 🙂

Reply:
I'd prefer to not have `NonX`, it ends up being awkward as it includes a negation itself. `NonX` is the same as `X` set to the other logical value, i.e flip true/false instead of inverting it through the name itself.

Reply:
@Sean-Der Any thoughts on the naming?

Reply:
Yea I think that would be nice!

I wouldn't have thought of this before, but https://refactoring.com/catalog/removeDoubleNegative.html and https://derickbailey.com/2014/03/27/double-negatives-in-code-dont-not-avoid-double-negative-boolean-logic/ have some good points on it also.

Reply:
Thank you for the suggestion! I went ahead and switched to "certificate" throughout the PR, and while I was changing `parseCipherSuites` I switched around the order in which the certificate and PSK cipher suites are checked to match up with the order of the parameters; it was a subtle thing I missed but I think it reads more nicely now!